discretize/base/base_mesh.py

```diff
@@ -124,288 +125,375 @@ def dim(self):
-        :return: normals, (sum(nF), dim)
+        Returns
+        -------
+        numpy.ndarray
+            normals, (n_faces, dim)
         """
         if self.dim == 2:
-            nX = np.c_[
-                np.ones(self.nFx), np.zeros(self.nFx)
-            ]
-            nY = np.c_[
-                np.zeros(self.nFy), np.ones(self.nFy)
-            ]
+            nX = np.c_[np.ones(self.n_faces_x), np.zeros(self.n_faces_x)]
+            nY = np.c_[np.zeros(self.n_faces_y), np.ones(self.n_faces_y)]
             return np.r_[nX, nY]
         elif self.dim == 3:
             nX = np.c_[
-                np.ones(self.nFx), np.zeros(self.nFx), np.zeros(self.nFx)
+                np.ones(self.n_faces_x),
+                np.zeros(self.n_faces_x),
+                np.zeros(self.n_faces_x),
             ]
             nY = np.c_[
-                np.zeros(self.nFy), np.ones(self.nFy), np.zeros(self.nFy)
+                np.zeros(self.n_faces_y),
+                np.ones(self.n_faces_y),
+                np.zeros(self.n_faces_y),
             ]
             nZ = np.c_[
-                np.zeros(self.nFz), np.zeros(self.nFz), np.ones(self.nFz)
+                np.zeros(self.n_faces_z),
+                np.zeros(self.n_faces_z),
+                np.ones(self.n_faces_z),
             ]
             return np.r_[nX, nY, nZ]
 
     @property
-    def tangents(self):
+    def edge_tangents(self):
         """Edge Tangents
 
-        :rtype: numpy.ndarray
-        :return: normals, (sum(nE), dim)
+        Returns
+        -------
+        numpy.ndarray
+            normals, (n_edges, dim)
         """
         if self.dim == 2:
-            tX = np.c_[
-                np.ones(self.nEx), np.zeros(self.nEx)
-            ]
-            tY = np.c_[
-                np.zeros(self.nEy), np.ones(self.nEy)
-            ]
+            tX = np.c_[np.ones(self.n_edges_x), np.zeros(self.n_edges_x)]
+            tY = np.c_[np.zeros(self.n_edges_y), np.ones(self.n_edges_y)]
             return np.r_[tX, tY]
         elif self.dim == 3:
             tX = np.c_[
-                np.ones(self.nEx), np.zeros(self.nEx), np.zeros(self.nEx)
+                np.ones(self.n_edges_x),
+                np.zeros(self.n_edges_x),
+                np.zeros(self.n_edges_x),
             ]
             tY = np.c_[
-                np.zeros(self.nEy), np.ones(self.nEy), np.zeros(self.nEy)
+                np.zeros(self.n_edges_y),
+                np.ones(self.n_edges_y),
+                np.zeros(self.n_edges_y),
             ]
             tZ = np.c_[
-                np.zeros(self.nEz), np.zeros(self.nEz), np.ones(self.nEz)
+                np.zeros(self.n_edges_z),
+                np.zeros(self.n_edges_z),
+                np.ones(self.n_edges_z),
             ]
             return np.r_[tX, tY, tZ]
 
-    def projectFaceVector(self, fV):
-        """Given a vector, fV, in cartesian coordinates, this will project
+    def project_face_vector(self, fV):
```

Comment:
this is minor, but would it be worth `fV` --> `face_vector` to be a bit more explicit in naming

Reply:
might as well just change it now